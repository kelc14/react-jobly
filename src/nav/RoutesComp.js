import React from "react";
import { Routes, Route } from "react-router-dom";

// components
import Home from "../home/Home";

import LoginForm from "../auth/LoginForm";
import SignUpForm from "../auth/SignUpForm";
import Companies from "../companies/Companies";
import CompanyDetails from "../companies/CompanyDetails";
import Jobs from "../jobs/Jobs";
import ProfileForm from "../profile/ProfileForm";
import ProtectedRoutes from "./ProtectedRoutes";

const RoutesComp = ({ loginUser, updateUser }) => {
  return (
    <Routes>
      <Route path="/login" element={<LoginForm loginUser={loginUser} />} />
      <Route path="/signup" element={<SignUpForm loginUser={loginUser} />} />

      <Route
        path="/companies"
        element={
          <ProtectedRoutes>
            <Companies />
          </ProtectedRoutes>
        }
      />
      <Route
        path="/companies/:handle"
        element={
          <ProtectedRoutes>
            =<CompanyDetails />
          </ProtectedRoutes>
        }
      />

      <Route
        path="/jobs"
        element={
          <ProtectedRoutes>
            <Jobs />
          </ProtectedRoutes>
        }
      />

      <Route
        path="/profile"
        updateUser={updateUser}
        element={
          <ProtectedRoutes updateUser={updateUser}>
            <ProfileForm updateUser={updateUser} />
          </ProtectedRoutes>
        }
      />

      <Route path="/" element={<Home />} />
    </Routes>
  );
};

export default RoutesComp;
