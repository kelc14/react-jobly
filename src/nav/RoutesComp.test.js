import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import RoutesComp from "./RoutesComp";

/**  SMOKE TEST */
it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <RoutesComp />
    </MemoryRouter>
  );
});

/**  SNAPSHOT TEST */
it("renders without crashing", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <RoutesComp />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
