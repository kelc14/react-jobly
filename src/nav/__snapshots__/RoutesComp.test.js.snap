// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<DocumentFragment>
  <h1
    class="Home-heading"
  >
    JOBLY
  </h1>
  <p
    class="Home-text"
  >
    All the jobs in one, convenient place.
  </p>
  <a
    class="Home-btn"
    href="/login"
  >
    Log in
  </a>
  <a
    class="Home-btn"
    href="/signup"
  >
    Sign up
  </a>
</DocumentFragment>
`;
